man/man8/zfs-load-key.8

```diff
@@ -166,6 +166,29 @@ will become one. Alternatively, the
 .Fl i
 flag may be provided to cause an encryption root to inherit the parent's key
 instead.
+.Pp
+Changing the user's key does not change the underlying master key used to
+encrypt the dataset.  Accordingly, if the master key has been compromised,
+both existing data and new data written after the
+.Nm zfs Cm change-key
+would be readable, so this command must not be relied upon to address a
+master key compromise.
+.Pp
+In the event of a master key compromise, ideally the drives should be securely
+erased to remove all the old data (which is readable using the compromised
+master key). A less secure option would be to re-encrypt the data on the same
+pool by creating new datasets and copying the data (e.g. using
+.Nm zfs Cm send
+|
+.Nm zfs Cm recv Ns
+).
+.Pp
+Currently, the previous user key is not overwritten on disk when this command
+is used.  The master key, encrypted with the previous user key, will be
+accessible on disk for an indeterminate length of time.  If the user's key has
+been compromised, the master key should be treated as compromised, so this
+command must not be relied upon to address a user key compromise.  This may be
+improved in the future.
```

Comment:
Let's leave off `This may be improved in the future`, which is true of almost everything.